include/engine/guidance/assemble_geometry.hpp

```diff
@@ -79,8 +83,14 @@ inline LegGeometry assembleGeometry(const datafacade::BaseDataFacade &facade,
     cumulative_distance += current_distance;
     // segment leading to the target node
     geometry.segment_distances.push_back(cumulative_distance);
+
+    std::vector<DatasourceID> forward_datasources;
+    facade.GetUncompressedDatasources(target_node.forward_packed_geometry_id, forward_datasources);
+
     geometry.annotations.emplace_back(
-        LegGeometry::Annotation{current_distance, target_node.forward_weight / 10.});
+        LegGeometry::Annotation{current_distance,
+                                target_node.forward_weight / 10.,
+                                forward_datasources[target_node.fwd_segment_position]});
```

Comment:
Hm maybe this is unrelated, but here and above the `.emplace_back(T{arg0, arg1, ..})` calls should probably be `.push_back(T{arg0, arg1, ...})` calls --- `push_back` has an rvalue overload.